### PR TITLE
Spidercoordinator: sync kubeadm-config event to trigger the status update

### PR DESCRIPTION
**Pull request by cyclinder**
#### What type of PR is this?

- release/bug 

**What this PR does / why we need it**:

we should watch the kubeadm-config configMap

**Which issue(s) this PR fixes**:

Fixes https://github.com/spidernet-io/spiderpool/issues/3290
Fixes https://github.com/spidernet-io/spiderpool/issues/3292

**Special notes for your reviewer**:

### CODE REVIEW

**Review by weizhoublue — DISMISSED** [review]

**Comment by weizhoublue** [comment]
I suggest to change to a new title that helps recognize what bug it is, in the release log

**Comment by ty-dc** [comment]
Does this PR only modify kubeadm-config? Changes to kube-controller-manager Pod cannot be synchronized?

**Review by ty-dc — DISMISSED** [review]

File: pkg/coordinatormanager/coordinator_informer.go [hunk 1]
```diff
@@ -319,12 +319,7 @@ func (cc *CoordinatorController) enqueueCoordinatorOnUpdate(oldObj, newObj inter
...
 	}
 
-	coord, err := cc.CoordinatorLister.Get(cc.DefaultCoordinatorName)
-	if err != nil {
-		return
-	}
-
-	coordCopy := coord.DeepCopy()
+	coordCopy := oldCoord.DeepCopy()
 	if reflect.DeepEqual(coordCopy.Status, newCoord.Status) {
```
Comment (Icarus9913) [inline pkg/coordinatormanager/coordinator_informer.go:323]: 为什么什么不直接用oldCoord.status，还要调用一下DeepCopy

File: pkg/coordinatormanager/coordinator_informer.go [hunk 2]
```diff
@@ -334,39 +329,48 @@ func (cc *CoordinatorController) enqueueCoordinatorOnUpdate(oldObj, newObj inter
...
+func (cc *CoordinatorController) enqueueCoordinatorOnConfigMapUpdated(oldObj, newObj interface{}) {
 	oldCm := oldObj.(*corev1.ConfigMap)
 	newCm := newObj.(*corev1.ConfigMap)
-	if newCm.Name != ciliumConfig {
-		return
-	}
+	if newCm.Name == ciliumConfig || newCm.Name == kubeadmConfigMap {
+		if cmp.Diff(oldCm.Data, newCm.Data) == "" {
+			return
+		}
```
Comment (Icarus9913) [inline pkg/coordinatormanager/coordinator_informer.go:351]: cmp.Diff用途是为了更好的看出Diff的区别在哪里吗？
如果不打印出来，单纯只判断的话，为什么不用标准库的reflec.DeepEqual反而引用第三方的依赖?

**Review by weizhoublue — APPROVED** [review]

**Review by ty-dc — APPROVED** [review]

**Pull request merged**